### DISCUSSION TITLE
DigitalCore > new announce tags (genre)

### DISCUSSION THREAD

**Discussion by DigitGram** [autobrr/autobrr#1778] [Q&A] [answered]
Hi there,

Indexer "DigitalCore" has new genre-tags in its iRC announce. See example below:
```
[2024-10-21 13:33:15] ENDOR: NEW TORRENT in Music/MP3 :: Optx-Not_To_Think-(LPRBAS225)-SINGLE-WEB-2024-PTC :: https://digitalcore.club/api/v1/torrents/download/xxxxxxx :: Genre: [Drum & Bass]
[2024-10-21 13:36:16] ENDOR: NEW TORRENT in Movies/1080p :: Meridian 1990 1080p BluRay x264-OFT :: https://digitalcore.club/api/v1/torrents/download/xxxxxxx :: Tags: [p2p,unrar] :: Genre: [Drama, Fantasy, Horror]
```

Is it possible with autobrr to create a filter with a genre-tag?

Thanks in advance!

**Comment by DigitGram** [comment]
maybe change DigitalCore.yaml to following?

```
pattern: 'NEW TORRENT in (.+) :: (.+) :: (https:\/\/.+\/).+\/([0-9a-zA-Z]+)(?: :: \[(.+)\])? :: Genre: \[(.+)\])?'
        vars:
          - category
          - torrentName
          - baseUrl
          - torrentId
          - tags
          - releaseTags
```

regex newb here. Any chance, that I can test that in my docker image?

**Reply by KyleSanderson** [reply]
> regex newb here. Any chance, that I can test that in my docker image?

check autobrr.com for custom definitions, just name the indexer something else as your test (I think?).

**Reply by DigitGram** [reply]
> check autobrr.com for custom definitions, just name the indexer something else as your test (I think?).

Thanks, will try that out.

So we have 4 different cases at irc-announce regarding newly added tags:
1. "Tags" first, then "Genre:"
`NEW TORRENT in Movies/1080p :: Meridian 1990 1080p BluRay x264-OFT :: https://digitalcore.club/api/v1/torrents/download/xxxxxxx :: Tags: [p2p,unrar] :: Genre: [Drama, Fantasy, Horror]`
2. only "Genre:"
3. only "Tags:"
4. no "Tags:", & no "Genre:"

I came up with this:
```
        pattern: 'NEW TORRENT in (.+) :: (.+) :: (https:\/\/.+\/).+\/([0-9a-zA-Z]+)(?: :: Tags: | :: Genre: )?(?:\[([0-9a-zA-Z ,\-&]+)\])?(?: :: Tags: | :: Genre: )?(?:\[([0-9a-zA-Z ,\-&]+)\])?'
        vars:
          - category
          - torrentName
          - baseUrl
          - torrentId
          - releaseTags
          - tags
```
Will test and report back.
Note: changed prior tags (i.e.: p2p,unrar) to releaseTags

**Comment by DigitGram** [comment, marked as answer]
new code here is working so far:
```
        pattern: 'NEW TORRENT in (.+) :: (.+) :: (https:\/\/.+\/).+\/([0-9a-zA-Z]+)(?: :: Tags: )?(?:\[([0-9a-zA-Z ,\-&]+)\])?(?: :: Genre: )?(?:\[([0-9a-zA-Z ,\-&]+)\])?'
        vars:
          - category
          - torrentName
          - baseUrl
          - torrentId
          - releaseTags
          - tags
```
